### PR TITLE
Generate mappers for NetworkConnectivityServiceConnectionPolicy

### PR DESCRIPTION
**Pull request by justinsb**
- **chore: generate mappers for NetworkConnectivityServiceConnectionPolicy**
- **autogen: generate networkconnectivity proto/grpc bindings**
- **chore: handle some edge cases in generator tooling**

### CODE REVIEW

**Comment by justinsb** [comment]
/assign @yuwenma 

Looks like there are 4 phases for a brand new resource:

1) types & CRD (no real exit criteria?)
2) mappers (should pass round trip tests)
3) controller & test data (should work with real GCP)
4) mocks (should pass tests with controller, mockgcp and realgcp should be the same)

This is (2).  I'm thinking that we might be able to combine 1 & 2, particularly because there's no exit criteria for (1)

File: pkg/controller/direct/networkconnectivity/refs.go
Comment (yuwenma) [inline pkg/controller/direct/networkconnectivity/refs.go:1]: qq: the commented code also look good, why do we not include them?

File: pkg/controller/direct/networkconnectivity/roundtrip_test.go
Comment (yuwenma) [inline pkg/controller/direct/networkconnectivity/roundtrip_test.go:1]: 👍

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

Looks great! Need to take a deeper look at the changes under tools/controllerbuilder. But I think we use that more as experimental code and it shall not block this PR.

**Pull request merged**